charts/karpenter/templates/provisioning.karpenter.sh_provisioners.yaml

```diff
@@ -52,6 +52,18 @@ spec:
                 - endpoint
                 - name
                 type: object
+              expiration:
```

Comment:
Will there be a way for a user to pause this or opt out specific instances? A use case I could think of is during a big event/launch it's nice to temporarily pause these automatic termination events to prevent potential outages. It would also be nice to have a label that can be added (or removed) to skip a specific instance from expiring if you need to do troubleshooting or debugging on it. If karpenter adds a label it would be nice for a company to know which nodes will be automatically recycled based on a common label.

Reply:
Some sort of locking mechanism at the cluster level would also be helpful. Something that understands the health of the cluster and a percentage of healthy instances to consider before blindly terminating instances. I've been in real world situations where automation was terminating healthy instances while new, unhealthy instances were being created which was quickly degrading the health of the system.

Reply:
Totally agree. I think we should respect a `node.karpenter.sh/do-not-expire` label on nodes. I see this as more of a manual intervention, so I'm wary of plumbing this through the API. I'll include this with the implementation.

Reply:
> Something that understands the health of the cluster and a percentage of healthy instances to consider before blindly terminating instances.
See @njtran's doc on node termination budgets, which handles exactly this.

Reply:
Further, if you remove the expiration concept from the provisioner spec, it will never expire nodes (even if the nodes were created).